### PR TITLE
Endpoint done

### LINKED ISSUE #3: feat: [User registration] add validations and return proper errors.
### Objective
Enhance the user creation process by implementing a series of input validations to ensure data integrity and provide clear feedback for invalid inputs.

Changes needs to be done in: `server\routes\users.js` for `/register` route, use postman to test.

### Validation Requirements

1. **Email Uniqueness**: Ensure that the user's email address is unique within the system. If a user attempts to register with an email that is already in use, the system should respond with the following error message:
   - Error: "User with email already exists."

2. **Phone Number Validation**: Verify that the phone number provided by the user is in a valid format. If the phone number does not meet the expected format, the system should return:
   - Error: "Invalid phone number."

3. **Name Length Check**: The user's name should adhere to a maximum length constraint to maintain data consistency. If the provided name exceeds this limit, the system should issue the following error:
   - Error: "Name too long."

### Error Response Structure

When errors are present, the response JSON should be structured as follows, including an array of error messages and a status within the `data` object indicating that the user was not created:

```json
{
  "errors": ["User with email already exists", "Invalid phone number"],
  "data": {
    "status": "User not created"
  }
}
```

### Successful Validation Response

For cases where all validations pass and the user is successfully created, the response JSON should contain an empty `errors` array and a `data` object with a success status:

```json
{
  "errors": [],
  "data": {
    "status": "User created successfully"
  }
}
```

### Implementation Notes

- Ensure that all error messages are user-friendly and accurately describe the validation failure.
- The final implementation should gracefully handle multiple validation errors, aggregating all relevant messages into the `errors` array for a single, comprehensive response.

### PR DESCRIPTION
**Pull request by Bashamega**
fixes #3

### CODE REVIEW

File: server/routes/users.js [hunk 1]
```diff
@@ -2,17 +2,104 @@ import express from 'express';
...
+async function UserExists(email){
+    try {
+        const user = await User.findOne({ where: { email } });
+        return user !== null;
+    } catch (error) {
+        console.error('Error checking if user exists:', error);
+        return false;
+    }
+};
+function validatePhoneNumber(input_str) {
```
Comment (lakshman-chaudhary) [inline server/routes/users.js]: let's create a new folder for these helpers and keep the users route file clean.
We can create validators.js under `server\utils\validators.js`.

File: server/routes/users.js [hunk 2]
```diff
@@ -2,17 +2,104 @@ import express from 'express';
...
+        if(phoneNumber){
+            if(!validatePhoneNumber(phoneNumber)){
+                errorsList.push('Invalid phone Number')
+            }
+        }else{
+            errorsList.push('Phone is required')
+        }
+
+        if(errorsList.length ==0){
+            // TODO: Add validations and return proper errors.
```
Comment (lakshman-chaudhary) [inline server/routes/users.js]: This TODO can be removed now.

File: server/routes/users.js [hunk 3]
```diff
@@ -2,17 +2,104 @@ import express from 'express';
...
+    try {
+        const user = await User.findOne({ where: { email } });
+        return user !== null;
+    } catch (error) {
+        console.error('Error checking if user exists:', error);
+        return false;
+    }
+};
+function validatePhoneNumber(input_str) {
+    var re = /^\(?(\d{3})\)?[- ]?(\d{3})[- ]?(\d{4})$/;
```
Comment (lakshman-chaudhary) [inline server/routes/users.js]:
```suggestion
    const re = /^\(?(\d{3})\)?[- ]?(\d{3})[- ]?(\d{4})$/;
```

**Review by lakshman-chaudhary — CHANGES_REQUESTED** [review]
Thanks for the efforts, minor suggestions.

**Comment by Bashamega** [comment]
Hello:)
I have updated it. Please review it now

File: server/routes/users.js [hunk 4]
```diff
@@ -1,18 +1,82 @@
...
+            }
+            if(await UserExists(email)){
+                errorsList.push('This Email is taken')
+            }
+
+        }else{
+            errorsList.push('Email address is required')
+        }
+        if(password){
+            var regex = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[$@#&!]).{8,}$/;
```
Comment (lakshman-chaudhary) [inline server/routes/users.js]:
```suggestion
            cont regex = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[$@#&!]).{8,}$/;
```

File: server/utils/validators.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    try {
+        const user = await User.findOne({ where: { email } });
+        return user !== null;
+    } catch (error) {
+        console.error('Error checking if user exists:', error);
+        return false;
+    }
+};
+export function validatePhoneNumber(input_str) {
+    var re = /^\(?(\d{3})\)?[- ]?(\d{3})[- ]?(\d{4})$/;
```
Comment (lakshman-chaudhary) [inline server/utils/validators.js]:
```suggestion
    const re = /^\(?(\d{3})\)?[- ]?(\d{3})[- ]?(\d{4})$/;
```

**Review by lakshman-chaudhary — CHANGES_REQUESTED** [review]

**Review by lakshman-chaudhary — APPROVED** [review]

**Comment by lakshman-chaudhary** [comment]
> Hello:) I have updated it. Please review it now

Thank you 🚀

**Comment by Bashamega** [comment]
> > Hello:) I have updated it. Please review it now
> 
> Thank you 🚀

Your welcome

**Comment by Bashamega** [comment]
When will you merge?

**Pull request merged**